src/libraries/Common/src/Interop/SunOS/procfs/Interop.ProcFs.Definitions.cs

```diff
@@ -0,0 +1,155 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+// C# equivalents for <sys/procfs.h> structures. See: struct lwpsinfo, struct psinfo.
+// We read directly onto these from procfs, so the layouts and sizes of these structures
+// must _exactly_ match those in <sys/procfs.h>
+
+// analyzer incorrectly flags fixed buffer length const
+// (https://github.com/dotnet/roslyn/issues/37593)
+#pragma warning disable CA1823
+
+internal static partial class Interop
+{
+    internal static partial class @procfs
+    {
+        internal const string RootPath = "/proc/";
+        private const string psinfoFileName = "/psinfo";
+        private const string lwpDirName = "/lwp";
+        private const string lwpsinfoFileName = "/lwpsinfo";
+
+        // Constants from sys/procfs.h
+        private const int PRARGSZ = 80;
+        private const int PRCLSZ = 8;
+        private const int PRFNSZ = 16;
+
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct @timestruc_t
+        {
+            public long tv_sec;
+            public long tv_nsec;
+        }
+
+        // lwp ps(1) information file.  /proc/<pid>/lwp/<lwpid>/lwpsinfo
+        // Equivalent to sys/procfs.h struct lwpsinfo
+        // "unsafe" because it has fixed sized arrays.
+        [StructLayout(LayoutKind.Sequential)]
+        internal unsafe struct @lwpsinfo
+        {
+            private     int     pr_flag;        /* lwp flags (DEPRECATED; do not use) */
+            public      uint    pr_lwpid;       /* lwp id */
+            private     long    pr_addr;        /* internal address of lwp */
+            private     long    pr_wchan;       /* wait addr for sleeping lwp */
+            public      byte    pr_stype;       /* synchronization event type */
+            public      byte    pr_state;       /* numeric lwp state */
+            public      byte    pr_sname;       /* printable character for pr_state */
+            public      byte    pr_nice;        /* nice for cpu usage */
+            private     short   pr_syscall;     /* system call number (if in syscall) */
+            private     byte    pr_oldpri;      /* pre-SVR4, low value is high priority */
+            private     byte    pr_cpu;         /* pre-SVR4, cpu usage for scheduling */
+            public      int     pr_pri;         /* priority, high value is high priority */
+            private     ushort  pr_pctcpu;      /* fixed pt. % of recent cpu time */
+            private     ushort  pr_pad;
+            public      timestruc_t pr_start;   /* lwp start time, from the epoch */
+            public      timestruc_t pr_time;    /* usr+sys cpu time for this lwp */
```

Comment:
If you implement my other suggestion, all fields can be private and be accessed by public properties.